### PR TITLE
Finally

### PR DESCRIPTION
**Pull request by Kangz**
PTAL, this is the actual fix for fbodepthbuffer and framebufferblit.depth_stencil.

Basically when doing a depth compare of 0.5 and 0.5 in the tests, what was actually happening was that we were comparing a depth of 0.5 for the fragment and 0.5 + epsilon for the depth buffer because it was quantized.

To fix this we convert the incoming fragment buffer depth to the depthbuffer format and back, making it incur the same loss of precision.

The rrUtil change is a noop cleanup I did thinking it might have been the root cause of this bug.

### CODE REVIEW

File: sdk/tests/deqp/framework/referencerenderer/rrFragmentOperations.js [hunk 1]
```diff
@@ -196,7 +196,18 @@ rrFragmentOperations.executeDepthCompare = function(inputFragments, depthFunc, d
                 var fragSampleNdx = 0;
                 var depthBufferValue = depthBuffer.getPixDepth(fragSampleNdx, frag.pixelCoord[0], frag.pixelCoord[1]);
                 var sampleDepthFloat = frag.sampleDepths[fragSampleNdx];
-                var sampleDepth = deMath.clamp(sampleDepthFloat, 0.0, 1.0);
+
+                /* convert input float to target buffer format for comparison */
```
Comment (Kangz) [inline sdk/tests/deqp/framework/referencerenderer/rrFragmentOperations.js]: Matches https://android.googlesource.com/platform/external/deqp/+/master/framework/referencerenderer/rrFragmentOperations.cpp#194

File: sdk/tests/deqp/framework/referencerenderer/rrUtil.js [hunk 2]
```diff
@@ -91,7 +88,17 @@ goog.scope(function() {
             ctx.bindBuffer(gl.ARRAY_BUFFER, null);
         }
 
-        ctx.drawQuads(gl.TRIANGLES, 0, 6);
+        {
```
Comment (Kangz) [inline sdk/tests/deqp/framework/referencerenderer/rrUtil.js]: Matches https://android.googlesource.com/platform/external/deqp/+/master/framework/opengl/simplereference/sglrContextUtil.cpp#97

**Comment by zhenyao** [comment]
This is great work!  LGTM
Let's merge and roll to see if this would cause any regressions in other tests (I reverted a CL on this a while ago due to causing a regression in another test).

File: sdk/tests/deqp/framework/referencerenderer/rrFragmentOperations.js [hunk 3]
```diff
@@ -196,7 +196,18 @@ rrFragmentOperations.executeDepthCompare = function(inputFragments, depthFunc, d
...
+                var access = new tcuTexture.PixelBufferAccess({
+                    format: depthBuffer.getFormat(),
+                    width: 1,
+                    height: 1,
+                    depth: 1,
+                    data: new ArrayBuffer(8)
+                });
```
Comment (kenrussell) [inline sdk/tests/deqp/framework/referencerenderer/rrFragmentOperations.js]: Could the allocation of the PixelBufferAccess object be hoisted outside this loop so allocations aren't performed per-pixel?

**Comment by kenrussell** [comment]
Awesome work Corentin! The optimization can wait until it's known whether this patch causes any regressions. Thanks for tracking this down!

**Comment by zhenyao** [comment]
@Kangz This seems to break fbodepthbuffer.html, but I think this is the right direction to move forward, so let's not revert this.  Can you take a look at the failure cases?

**Comment by Kangz** [comment]
Which platforms does it break? It passed up to driver bugs on NVIDIA Linux.

**Comment by zhenyao** [comment]
On Win AMD / NVidia.  Mac bots aren't running, but at least it's failing on my Mac AMD.

**Comment by Kangz** [comment]
See #1856 for a follow up fix.

**Pull request merged**